server/etcdserver/api/etcdhttp/health.go

```diff
@@ -193,3 +197,171 @@ func checkAPI(ctx context.Context, lg *zap.Logger, srv ServerHealth, serializabl
 	lg.Debug("serving /health true")
 	return h
 }
+
+type HealthCheck func(ctx context.Context) error
+
+type CheckRegistry struct {
+	path   string
+	checks map[string]HealthCheck
+}
+
+func installLivezEndpoints(lg *zap.Logger, mux *http.ServeMux, server ServerHealth) {
+	reg := CheckRegistry{path: "/livez", checks: make(map[string]HealthCheck)}
+	reg.Register("serializable_read", serializableReadCheck(server))
```

Comment:
QQ, doesn't serializable read require store not to be corrupted?

Reply:
It doesn't help to restart etcd on data corruption, so data corruption isn't a check for livez.

But if etcd (bbolt) crashes due to data corruption, then it's another story. It needs human intervention anyway in that case.

Reply:
I'm not asking "Why there is no corrupt_alarm check?". I'm asking, if etcd cluster is corrupted, I expect every etcd member to return error on serializable read. Meaning that serializable_read is also checking corrupt alarm. 

I will double check the behavior.

Reply:
Oh, wait. I was thinking about corrupt applier. Might not be the case as we read doesn't go through checking corrupt. Hmm, but this might also be a weird behavior.

Reply:
Ok, wrote a PR that tests livez readyz and confirms that livez is not triggered by corruption. https://github.com/etcd-io/etcd/pull/16784